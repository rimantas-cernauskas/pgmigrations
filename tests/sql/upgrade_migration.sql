-- Test function pgmigrations.rollback_migration
\i tests/sql/setup/init.sql
BEGIN;
SELECT plan(7);
SET ROLE __pgmigrations_testuser;

-- Setup migration scripts
SET ROLE __pgmigrations_superuser;
SELECT is(
    pgmigrations.add_migration(
         '00'
        ,'CREATE TABLE table0( id SERIAL );'
        ,'DROP TABLE table0;'
    )
    ,TRUE
    ,'should add migration "00"'
);

SELECT is(
    pgmigrations.add_migration(
         '01'
        ,'CREATE TABLE table1( id SERIAL );'
        ,'DROP TABLE table1;'
    )
    ,TRUE
    ,'should add migration "01"'
);

-- upgrade first migration
SELECT is(
     pgmigrations.upgrade_migration( '00' )
    ,TRUE
    ,'upgrade_migration() should return TRUE on successful run'
);

-- check that migration was actually executed
-- migration 00
SELECT has_table('table0');
SELECT has_column('table0', 'id');

-- and that 01 migration was not executed
SELECT hasnt_table( 'table1' );

-- check log
PREPARE my_query AS
    SELECT
         migration
        ,up_run
        ,up_success
        ,up_error
    FROM pgmigrations.migrations_info
    ORDER BY migration ASC;

PREPARE my_expect AS
    SELECT * FROM (
        VALUES (
             '00'
            ,NOW()
            ,TRUE
            ,NULL
        ),
        (
            '01'
            ,NULL
            ,NULL
            ,NULL
        )
    )
    t(
         migration
        ,down_run
        ,dwon_success
        ,down_error
    )
;

SELECT results_eq(
     'my_query'
    ,'my_expect'
    ,'should get expected migrations info'
);
DEALLOCATE my_query;
DEALLOCATE my_expect;

SELECT * FROM finish();
ROLLBACK;