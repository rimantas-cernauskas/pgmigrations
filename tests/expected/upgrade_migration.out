-- Test function pgmigrations.rollback_migration
\i tests/sql/setup/init.sql
\set ECHO errors
1..7
ok 1 - should add migration "00"
ok 2 - should add migration "01"
ok 3 - upgrade_migration() should return TRUE on successful run
ok 4 - Table table0 should exist
ok 5 - Column table0.id should exist
ok 6 - Table table1 should not exist
ok 7 - should get expected migrations info
